packages/runtime/src/useSSRComputation_Client.tsx

```diff
@@ -1,22 +1,42 @@
 import { useEffect, useMemo, useState } from "react";
-import { useSSRCache } from "./SSRCacheProvider";
 import { calculateCacheKey, ClientFunction, Dependency, parseDependencies } from "./utils";
 import { wrapErrorHandler } from "./errorHandler";
+import { useAsyncResultManager } from "./useAsyncResultManager";
+import { getSSRCache } from "./ssrCache";
+import { runOnSubscriptionsResumed } from "./subscriptions";
 
 const useSSRComputation_Client: ClientFunction = (importFn, dependencies, options, relativePathToCwd) => {
   const [fn, setFn] = useState<(...dependencies: Dependency[])=>any>();
-  const [, forceUpdate] = useState(0);
-  const cache = useSSRCache();
+  const [forceExecution, setForceExecution] = useState(false);
+  const cache = getSSRCache();
   const parsedDependencies = parseDependencies(dependencies);
   const skip = !!options.skip;
 
   // relativePathToCwd is used to make sure that the cache key is unique for each module
   // and it's not affected by the file that calls it
   const cacheKey = calculateCacheKey(relativePathToCwd, parsedDependencies);
-  const isCacheHit = cache?.[cacheKey];
+  const cachedValue = cache?.[cacheKey]?.result;
+  const isSubscription = cache?.[cacheKey]?.isSubscription || false;
+  const isCacheHit = !!cachedValue;
 
   useEffect(() => {
-    if (isCacheHit || skip) return;
+    let isMounted = true;
+
+    if (isSubscription) {
+      void runOnSubscriptionsResumed(() => {
+        if (isMounted) {
+          setForceExecution(true);
```

Comment:
This call will generate an avoidable re-render and an avoidable re-execution of the `fn` function.
Both re-render and re-execution are potentially CPU-heavy tasks. And they can be avoided on the level of the plugin.

When subscriptions are resumed instead of updating the React state write an async function and keep the state as a simple variable (outside of react State).

I'm talking of something like:
```
  useEffect(() => {
    let isMounted = true;
    let unsubscribe = () => {};

    if (isSubscription) {
      void runOnSubscriptionsResumed( async () => {
        if (!isMounted) return;
        const subscribe = (await importFn()).subscribe;
        if (!isMounted) return;
        unsubscribe = subscribe({
          next: (value) => {
            if (!isMounted) return;
            setSubscriptionResult(value);
          }
        });
      });
    }

    return () => {
      unsubscribe();
      isMounted = false;
    }
  }, [isSubscription]);
```
If we implement it this way, we will not cause expensive re-rendering until the client calls `next` function.